### DISCUSSION TITLE
back_populates on relationship x refers to attribute y that is not a relationship

### DISCUSSION THREAD

**Discussion by greenseeker** [fastapi/sqlmodel#932] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
class Cids(rx.Model, table=True):
    cid: str = Field(primary_key=True)
    company: str

    tickets: list["Servers"] = Relationship(back_populates="ticket_number")  

class Servers(rx.Model, table=True):
    cid: str = Field(foreign_key="cids.cid")
    ticket_number: str
    instance: int
    owner: str
    template: str
    power_state: int = 0
    note: str | None = None

    company: Cids = Relationship(back_populates="company")
```

### Description

I'm doing this in reflex which incorporates SQLModel, but there are slight differences (eg. rx.Model in class declarations) but they shouldn't be relevant to the issue.

I just can't quite wrap my head around declaring relationships and all the docs and examples I've looked at haven't made it click.

I've got 2 tables: one for Company IDs (Cids) and one for Servers. The Cids table is just a list of IDs with the actual company name. The Servers list contains the Company ID but I want to display the company name in the app, so I've got to do a join. Each server can only have a single company, but each company can have effectively unlimited servers.

Presently when I try to start my app with the declarations above, I get:
> One or more mappers failed to initialize - can't proceed with initialization of other mappers. Triggering mapper: 'Mapper[Cids(cids)]'. Original exception was: back_populates on relationship 'Cids.tickets' refers to attribute 'Servers.ticket_number' that is not a relationship.  The back_populates parameter should refer to the name of a relationship on the target class.

### Operating System

Linux

### SQLModel Version

0.0.16

### Python Version

3.11.6

**Comment by joachimhuet** [comment, marked as answer]
Hey, [back_populates](https://docs.sqlalchemy.org/en/20/orm/relationship_api.html#sqlalchemy.orm.relationship.params.back_populates) refers to the name of the relationship in the related class.

So in your case you should have something like that:
```python
class Cids(rx.Model, table=True):
    cid: str = Field(primary_key=True)
    company: str

    tickets: list["Servers"] = Relationship(back_populates="company")  

class Servers(rx.Model, table=True):
    cid: str = Field(foreign_key="cids.cid")
    ticket_number: str
    instance: int
    owner: str
    template: str
    power_state: int = 0
    note: str | None = None

    company: Cids = Relationship(back_populates="tickets")
```
